config/detekt/detekt-baseline.xml

```diff
@@ -13,7 +13,7 @@
     <ID>LongParameterList:Backend.kt$Backend$( purchaseToken: String, appUserID: String, productID: String, isRestore: Boolean, offeringIdentifier: String?, observerMode: Boolean, price: Double?, currency: String?, subscriberAttributes: Map&lt;String, SubscriberAttribute&gt;, onSuccess: PostReceiptDataSuccessCallback, onError: PostReceiptDataErrorCallback )</ID>
     <ID>LongParameterList:EntitlementInfo.kt$EntitlementInfo$( val identifier: String, val isActive: Boolean, val willRenew: Boolean, val periodType: PeriodType, val latestPurchaseDate: Date, val originalPurchaseDate: Date, val expirationDate: Date?, val store: Store, val productIdentifier: String, val isSandbox: Boolean, val unsubscribeDetectedAt: Date?, val billingIssueDetectedAt: Date? )</ID>
     <ID>LongParameterList:PurchaserInfo.kt$PurchaserInfo$( val entitlements: EntitlementInfos, val purchasedNonSubscriptionSkus: Set&lt;String&gt;, val allExpirationDatesByProduct: Map&lt;String, Date?&gt;, val allPurchaseDatesByProduct: Map&lt;String, Date?&gt;, @Deprecated("Use getExpirationDateForEntitlement instead") val allExpirationDatesByEntitlement: Map&lt;String, Date?&gt;, @Deprecated("Use getPurchaseDateForEntitlement instead") val allPurchaseDatesByEntitlement: Map&lt;String, Date?&gt;, val requestDate: Date, internal val jsonObject: JSONObject, internal val schemaVersion: Int, val firstSeen: Date, val originalAppUserId: String )</ID>
-    <ID>LongParameterList:Purchases.kt$Purchases$( private val applicationContext: Context, backingFieldAppUserID: String?, private val backend: Backend, private val billingWrapper: BillingWrapper, private val deviceCache: DeviceCache, private val executorService: ExecutorService, private val identityManager: IdentityManager, private val subscriberAttributesManager: SubscriberAttributesManager, appConfig: AppConfig )</ID>
```

Comment:
detekt was complaining about a long parameter list, so I had to run 
`./gradlew detektAllBaseline` again since I don't think we should change this now.

Reply:
we'll eventually have to move to a configuration object or something if this list gets too crazy, but happy to postpone it